File: frontend/src/types/system.ts
export interface SystemResponse {
  id: number;
  name: string;
  description: string | null;
  tenant_id: number;
  github_repository_url: string | null;
  custom_fields: Record<string, unknown> | null;
  created_at: string;
  updated_at: string;
}

export interface SubSystemResponse {
  id: number;
  name: string;
  description: string | null;
  system_id: number;
  tenant_id: number;
  component_type: string;
  technology: string | null;
  custom_fields: Record<string, unknown> | null;
  created_at: string;
  updated_at: string;
}

export interface SystemCreate {
  name: string;
  description?: string;
  github_repository_url?: string;
  custom_fields?: Record<string, unknown>;
}

export interface SystemUpdate {
  name?: string;
  description?: string;
  github_repository_url?: string;
  custom_fields?: Record<string, unknown>;
}

export interface SubSystemCreate {
  name: string;
  description?: string;
  custom_fields?: Record<string, unknown>;
}

export interface SubSystemUpdate {
  name?: string;
  description?: string;
  custom_fields?: Record<string, unknown>;
}

File: backend/app/api/v1/schemas/system.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import datetime

from app.db.models.system import ComponentType


class SystemCreate(BaseModel):
    name: str
    description: Optional[str] = None
    github_repository_url: Optional[str] = None
    custom_fields: Optional[dict] = None


class SystemUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    github_repository_url: Optional[str] = None
    custom_fields: Optional[dict] = None


class SystemResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    description: Optional[str] = None
    tenant_id: int
    github_repository_url: Optional[str] = None
    custom_fields: Optional[dict] = None
    created_at: datetime
    updated_at: datetime


class SubSystemCreate(BaseModel):
    name: str
    description: Optional[str] = None
    component_type: ComponentType = ComponentType.OTHER
    technology: Optional[str] = None
    custom_fields: Optional[dict] = None


class SubSystemUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    component_type: Optional[ComponentType] = None
    technology: Optional[str] = None
    custom_fields: Optional[dict] = None


class SubSystemResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    description: Optional[str] = None
    component_type: ComponentType
    technology: Optional[str] = None
    system_id: int
    tenant_id: int
    custom_fields: Optional[dict] = None
    created_at: datetime
    updated_at: datetime
